File: spring/webflux-kotlin/src/main/java/de/welcz/hello/GreetingWebClient.java
package de.welcz.hello;

import org.springframework.http.MediaType;
import org.springframework.web.reactive.function.client.ClientResponse;
import org.springframework.web.reactive.function.client.WebClient;
import reactor.core.publisher.Mono;

class GreetingWebClient {
    private WebClient client = WebClient.create("http://localhost:8080");

    private Mono<ClientResponse> result = client.get()
                                                .uri("/hello")
                                                .accept(MediaType.TEXT_PLAIN)
                                                .exchange();

    String getResult() {
        return ">> result = " + result.flatMap(response -> response.bodyToMono(String.class)).block();
    }
}

File: spring/webflux-kotlin/src/main/kotlin/de/welcz/hello/GreetingWebClient.kt
package de.welcz.hello

import org.springframework.http.MediaType
import org.springframework.web.reactive.function.client.ClientResponse
import org.springframework.web.reactive.function.client.WebClient
import reactor.core.publisher.Mono

internal class GreetingWebClient {
    private val client = WebClient.create("http://localhost:8080")

    private val result = client.get()
            .uri("/hello")
            .accept(MediaType.TEXT_PLAIN)
            .exchange()

    fun retrieveResult(): String {
        return ">> result = " + result.flatMap { response -> response.bodyToMono(String::class.java) }.block()!!
    }
}
